src/colorizer/AnnotationData.ts

```diff
@@ -10,18 +10,32 @@ const CSV_COL_ID = "ID";
 const CSV_COL_TIME = "Frame";
 const CSV_COL_TRACK = "Track";
 
+export const BOOLEAN_VALUE = "TRUE";
+
 export const DEFAULT_ANNOTATION_LABEL_COLORS = KNOWN_CATEGORICAL_PALETTES.get(
   DEFAULT_CATEGORICAL_PALETTE_KEY
 )!.colorStops;
 
+export enum LabelType {
+  BOOLEAN = "boolean",
+  INTEGER = "integer",
+  CUSTOM = "custom",
+}
+
 export type LabelOptions = {
+  type: LabelType;
   name: string;
   color: Color;
+  autoIncrement: boolean;
 };
 
 export type LabelData = {
   options: LabelOptions;
+  valueToIds: Map<string, Set<number>>;
```

Comment:
I've left a couple comments on this in the code, but this makes the lookup for what values exist super simple, but make the lookup of a value from an ID potentially O(N). If we're having to do that operation a lot (like during rendering?) I may update the data model to include an `id->value` lookup.

Reply:
when do you need to do this kind of lookup?

Reply:
The `id -> value` lookup is used in rendering, like if I want to show an annotation label onscreen and have it render the value (for example, an integer), I need to do a lookup from the ID to the value that's assigned to it. The inverse, `value -> id`, is more useful for deciding whether a value is still assigned to IDs.